File: src/leetcode/2026/04/Solution830.java
import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;

public class Solution830 {
    public List<List<Integer>> largeGroupPositions(String s) {
        String[] groups = s.split("(?<=(.))(?!\\1)");
        List<List<Integer>> result = new ArrayList<>();
        int index = 0;
        for (String group : groups) {
            if (group.length() >= 3) {
                result.add(Arrays.asList(index, index + group.length() - 1));
            }
            index += group.length();
        }
        return result;
    }

    public static void main(String[] args) {
        Solution830 solution = new Solution830();

        String s1 = "abbxxxxzzy";
        List<List<Integer>> result1 = solution.largeGroupPositions(s1);
        System.out.println(result1); // Output: [[3, 6]]

        String s2 = "abc";
        List<List<Integer>> result2 = solution.largeGroupPositions(s2);
        System.out.println(result2); // Output: []

        String s3 = "abcdddeeeeaabbbcd";
        List<List<Integer>> result3 = solution.largeGroupPositions(s3);
        System.out.println(result3); // Output: [[3, 5], [6, 9], [12, 14]]
    }
}

File: src/leetcode/2026/04/Solution830.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

class Solution830 {
public:
    vector<vector<int>> largeGroupPositions(string s) {
        vector<vector<int>> result;
        int n = s.size();
        int i = 0;
        while (i < n) {
            int j = i;
            while (j < n && s[j] == s[i]) {
                j++;
            }
            if (j - i >= 3) {
                result.push_back({i, j - 1});
            }
            i = j;
        }
        return result;
    }
};

int main() {
    Solution830 solution;

    string s = "abbxxxxzzy";
    vector<vector<int>> result = solution.largeGroupPositions(s);
    for (const auto& group : result) {
        cout << "[" << group[0] << ", " << group[1] << "]" << endl;
    }

    string s2 = "abc";
    vector<vector<int>> result2 = solution.largeGroupPositions(s2);
    for (const auto& group : result2) {
        cout << "[" << group[0] << ", " << group[1] << "]" << endl;
    }

    string s3 = "abcdddeeeeaabbbcd";
    vector<vector<int>> result3 = solution.largeGroupPositions(s3);
    for (const auto& group : result3) {
        cout << "[" << group[0] << ", " << group[1] << "]" << endl;
    }


    return 0;
}